M2/Macaulay2/packages/ResolutionsOfStanleyReisnerRings.m2

```diff
@@ -0,0 +1,592 @@
+        Example
+		        S = QQ[a..e];
+            F = {a*b*c,c*d,e}
+            D = simplicialComplex F
+            colorfulPresentation D 
+            M = colorfulPresentation F 
+            degrees M 
+    SeeAlso
+        Module 
+        kSOP
+///
+
+-------------------
+-- Tests
+-------------------
+
+--kSOP,uSOP,colorfulPresentation,universalPresentation,barycentricSubdivision,sumBetti,equalBettiTally
+
+-- Tests barycentricSubdivision 
+TEST ///
+    S = QQ[a,b,c];
+    B = barycentricSubdivision simplicialComplex {a*b,c};
+    R = ring(B)
+    assert(B === simplicialComplex  {v_0*v_2*v_4, v_0*v_1*v_4, v_0*v_3});
+    S = QQ[a..e];
+    B = barycentricSubdivision simplicialComplex {a*b*c,c*d,e};
+    R = ring(B)
+    assert(B === simplicialComplex  {v_0*v_3*v_8*v_10,v_0*v_2*v_8*v_10,v_0*v_3*v_7*v_10,v_0*v_1*v_7*v_10,v_0*v_2*v_6*v_10,v_0*v_1*v_6*v_10,v_0*v_4*v_9,v_0*v_3*v_9,v_0*v_5});
+///
+
+-- Tests uSOP
+TEST ///
+    S = QQ[a,b,c];
+    assert(uSOP simplicialComplex {a*b,c} === {a+b+c,a*b});
+    S = QQ[a..e];
+    assert(uSOP simplicialComplex {a*b*c,c*d,e} === {a+b+c+d+e,a*b+a*c+b*c+c*d,a*b*c});
+///
+
+-- Tests kSOP
+TEST ///
+   S         = QQ[a,b,c];
+   B0        = barycentricSubdivision(simplicialComplex {a*b,c});
+   STrivial  = ring(B0);
+   B         = faceDelete(STrivial_0,B0);
+   assert(kSOP(B,3) === {v_1+v_2+v_3,v_4});
+   S         = QQ[a..e];
+   B0        = barycentricSubdivision(simplicialComplex {a*b*c,c*d,e});
+   STrivial  = ring(B0);
+   B         = faceDelete(STrivial_0,B0);
+   assert(kSOP(B,5) === {v_1+v_2+v_3+v_4+v_5,v_6+v_7+v_8+v_9,v_10});
+///
+
+-- Tests equalBettiTally 
+TEST ///
+    S = QQ[a..f];
+    m = matrix{{a,b,d,e},{b,c,e,f}}
+    n = matrix{{a,b},{c,d},{e,f}}
+    assert(equalBettiTally(coker m,coker n) === false);
+    n = matrix{{a,b,d,e},{b,c,e,f}}
+    assert(equalBettiTally(coker m,coker n));
+///
+end
+
+-- Tests sumBetti
+TEST ///
+   S = QQ[a..f];
+   assert(sumBetti(coker matrix{{a,b,d,e},{b,c,e,f}}) === 12);
+   assert(sumBetti(coker matrix{{a,b},{c,d},{e,f}} === 5);
+   S = QQ[a,b,c];
+   assert(sumBetti(coker matrix{{a},{b}}) == 3);
+///
+
+
+-- Tests universalPresentation
+TEST ///
+   S = QQ[a,b,c];
+   D = simplicialComplex {a*b,c} 
+   M = universalPresentation D;
+   f = map(target generators M, source relations M, matrix{{0},{0},{c}})
+   assert(relations M === f);
+///
+
+
+-- Tests colorfulPresentation
+TEST ///
+   S = QQ[a,b,c];
+   D = simplicialComplex {a*b,c} 
+   M = colorfulPresentation D;
+   f = map(target generators M, source relations M, matrix{{0},{0},{c}})
+   assert(relations M === f);
+///
+
```

Comment:
The documentation of `Kolors` is missing.

Reply:
removed it.